src/main/java/org/folio/services/kafka/topic/KafkaAdminClientService.java

```diff
@@ -37,30 +38,36 @@ public KafkaAdminClientService(Supplier<KafkaAdminClient> clientFactory) {
     this.clientFactory = clientFactory;
   }
 
-  public Future<Void> createKafkaTopics() {
+  public Future<Void> createKafkaTopics(String tenantId, String environmentName) {
     final KafkaAdminClient kafkaAdminClient = clientFactory.get();
-    return createKafkaTopics(kafkaAdminClient).onComplete(result -> {
-      if (result.succeeded()) {
-        log.info("Topics created successfully");
-      } else {
-        log.error("Unable to create topics", result.cause());
-      }
-
-      kafkaAdminClient.close().onComplete(closeResult -> {
-        if (closeResult.failed()) {
-          log.error("Failed to close kafka admin client", closeResult.cause());
+    return createKafkaTopics(tenantId, environmentName, kafkaAdminClient)
+      .onComplete(result -> {
+        if (result.succeeded()) {
+          log.info("Topics created successfully");
+        } else {
+          log.error("Unable to create topics", result.cause());
         }
-      });
+
+        kafkaAdminClient.close().onComplete(closeResult -> {
+          if (closeResult.failed()) {
+            log.error("Failed to close kafka admin client", closeResult.cause());
+          }
+        });
     });
   }
 
-  private Future<Void> createKafkaTopics(KafkaAdminClient kafkaAdminClient) {
-    final List<NewTopic> newTopics = readTopics();
+  private Future<Void> createKafkaTopics(String tenantId, String environmentName,
+    KafkaAdminClient kafkaAdminClient) {
+
+    final List<NewTopic> expectedTopics = readTopics()
+      .map(topic -> prefixWith(tenantId, topic))
+      .map(topic -> prefixWith(environmentName, topic))
```

Comment:
Maybe do a string concat in one call?

Reply:
I deliberately kept them separate to try to emphasise the separate aspects of the name qualification. 

I'll change this to a single string statement.